.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,30 @@
+name: Integration Tests
+
+on:
+    - pull_request
+    - push
+
+jobs:
+  dispatch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get the branch name
+        id: get_branch
+        run: echo ::set-output name=branch::$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}})
+
+      - name: Get the originating repository
+        id: get_repo
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            echo ::set-output name=repo::${{ github.event.pull_request.head.repo.full_name }}
+          else
+            echo ::set-output name=repo::${{ github.repository }}
+          fi
+
+      - name: Repository Dispatch
+        uses: peter-evans/repository-dispatch@ff45666b9427631e3450c54a1bcbee4d9ff4d7c0 # v3.0.0
```

Comment:
I guess this action is doing only a workflow dispatch? The corresponding `curl` would be something close to

```
payload='{
            "ref": "${{ env.TARGET_BRANCH }}",
            "inputs": {
            }
          }'

          curl -L \
            -X POST \
            -H "Accept: application/vnd.github+json" \
            -H "Authorization: Bearer ${{ secrets.bearer }}"\
            -H "X-GitHub-Api-Version: 2022-11-28" \
            https://api.github.com/repos/YOUR_ORG/YOUR_REPO/actions/workflows/YOUR_ACTION.yaml/dispatches \
            -d "$payload"
```